tests/system/common.py

```diff
@@ -88,6 +88,19 @@ def wait_for_mesos_endpoint(timeout_sec=timedelta(minutes=5).total_seconds()):
     return shakedown.time_wait(lambda: shakedown.mesos_available_predicate(), timeout_seconds=timeout_sec)
 
 
+def wait_for_metronome(timeout_sec=timedelta(minutes=5).total_seconds()):
+    """ Waits for the Metronome API url to be available for a given timeout.
+        It needs to succeed as many times as we have masters because that means that all adminrouters on all masters
+        are probably aware of the restarted leader.
+        Without this in multimaster setup this can be very flaky.
+    """
+
+    master_count = len(shakedown.get_all_masters())
+    shakedown.time_wait(lambda: metronome_available_predicate(),
```

Comment:
I'd much rather use retry with assert since it is easier to debug an `AssertException: assert 2 == 3` than a `shakedown.TimeoutException`. See [MARATHON-8077](https://jira.mesosphere.com/browse/MARATHON-8077). I'm not too strong on asserts but find it the easiest way.

Reply:
we use shakedown.. and there hasn't been a stated direction to move away from it.   If there is an issue... shouldn't we fix the issue with shakedown?

Reply:
Well, that's another debate from what I perceive. There is also dcos-test-utils which ported some things from shakedown and is not dependent on the dcos-cli as Shakedown is.

In any case, you raise a valid point. Where should we add improvements to our error reporting?

Reply:
I see your point. However, I'd like to avoid going down a road we may not want to be on. What's the benefit aside from consistency? I think there is a better alternative right now without much effort.